apollo-router/src/services/supergraph_service.rs

```diff
@@ -293,6 +293,21 @@ async fn service_call(
                     let (subs_tx, subs_rx) = mpsc::channel(1);
                     let query_plan = plan.clone();
                     let execution_service_factory_cloned = execution_service_factory.clone();
+                    let mut cloned_supergraph_req = SupergraphRequest::builder()
+                        .extensions(req.supergraph_request.body().extensions.clone())
+                        .and_query(req.supergraph_request.body().query.clone())
+                        .context(context.clone())
+                        .method(req.supergraph_request.method().clone())
+                        .and_operation_name(req.supergraph_request.body().operation_name.clone())
+                        .uri(req.supergraph_request.uri().clone())
+                        .variables(req.supergraph_request.body().variables.clone());
+
+                    for (header_name, header_value) in req.supergraph_request.headers().clone() {
+                        if let Some(header_name) = header_name {
+                            cloned_supergraph_req =
+                                cloned_supergraph_req.header(header_name, header_value);
+                        }
+                    }
```

Comment:
Maybe make this into a function and call the function at line 533?

Reply:
I wanted to make a function here but, it requires to clone the context everytime because we're using it before, so I didn't want to create a clone for every operations instead of only for subscription

Reply:
Fixed in https://github.com/apollographql/router/pull/4057/commits/c15c60286fab0528970ccadf1b6579f12447b5dc